internal/pkg/gateway/registry.go

```diff
@@ -165,12 +166,22 @@ func (reg *registry) endorsersByOrg(channel string, chaincode string) map[string
 	defer reg.configLock.RUnlock()
 
 	for _, member := range members {
+		pkiid := member.PKIid
+		// TODO use member.PreferredEndpoint() or go back to member.Endpoint?
 		endpoint := member.PreferredEndpoint()
+
 		// find the endorser in the registry for this endpoint
 		var endorser *endorser
-		if endpoint == reg.localEndorser.address {
+		if bytes.Equal(pkiid, reg.localEndorser.pkiid) {
+			logger.Debugw("Found local endorser", "pkiid", pkiid)
 			endorser = reg.localEndorser
+		} else if endpoint == "" {
+			// TODO the endpoint may be empty because discovery!!
+			// include seperate check/warning or just fall through to else???
+			reg.logger.Warnf("Failed to find endorser: no endpoint from discovery")
```

Comment:
Keep the warning in.  We'd need to understand why we had an endorser with no endpoint if it's not the local peer.  Might be worth logging the pkiid of this endorser.